ecommerce/views/v0/__init__.py

```diff
@@ -84,3 +90,49 @@ def get_queryset(self):
             .select_related("content_type")
             .prefetch_related("purchasable_object")
         )
+
+
+class BasketViewSet(NestedViewSetMixin, ModelViewSet):
+    """API view set for Basket"""
+    queryset = Basket.objects.all()
+    serializer_class = BasketSerializer
+    permission_classes = [IsAuthenticated]
+
+    def update(self, request, *args, **kwargs):
+        basket = self.get_object()
+        product_id = request.data.get("product_id")
+        serializer = BasketSerializer(basket)
+        return Response(serializer.data)
+
+
+class BasketItemViewSet(NestedViewSetMixin, ModelViewSet):
+    """API view set for BasketItem"""
+    queryset = BasketItem.objects.all()
+    serializer_class = BasketItemSerializer
+    permission_classes = [IsAuthenticated]
+
+
+@api_view(["POST"])
+@permission_classes([IsAuthenticated])
+def add_product_to_basket_view(request):
```

Comment:
I don't think this would require a separate view - it should be part of `BasketItemViewSet` I think.